server/src/main/kotlin/fi/oph/kitu/yki/YkiScheduledTasks.kt

```diff
@@ -0,0 +1,19 @@
+package fi.oph.kitu.yki
+
+import com.github.kagkarlsson.scheduler.task.Task
+import com.github.kagkarlsson.scheduler.task.helper.Tasks
+import com.github.kagkarlsson.scheduler.task.schedule.Schedules
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import java.time.Duration
+
+@Configuration
+@ConditionalOnProperty(name = ["kitu.scheduling.yki.enabled"], matchIfMissing = false)
+class YkiScheduledTasks {
+    @Bean
+    fun dailyImport(ykiService: YkiService): Task<Void> =
+        Tasks
+            .recurring("YKI-import-daily", Schedules.fixedDelay(Duration.ofSeconds(10L)))
```

Comment:
Pystytäänkö kutsu tekemään HTTP:n yli, vai onko käytännöllisempää tehdä kutsu suoraan serviceen?

Mietin, miten synkata sun työ mun työhön (https://github.com/Opetushallitus/koto-rekisteri/pull/137)

Reply:
Ei välttämättä ole hirveästi lisäarvoa tehdä API-kutsua ihan vaan HTTP-kutsun tekemisen ilosta. Jos mennään tällä lähestymistavalla, että scheduler asuu samassa monoliitissa itse palveluiden kanssa, ei mielestäni ole hirveästi iloa siitä, että eräajot käynnisteltäisiin HTTP-pyynnöillä. Ne API-endpointit eräajojen käynnistelyyn on ihan kivoja jos niitä tarvitsee koskaan käynnistellä manuaalisesti, mutta HTTP-pyynnöt järjestelmän sisällä ihan vaan tuon pakottamista varten kuulostaa vähän omituiselta patternilta.

Näköjään testaillessa myös jäänyt tuo aikataulu suorittamaan tuo esimerkkieräajo 10 sekunnin välein eikä päivittäin.